test/sanity/code-smell/docs-build.py

```diff
@@ -14,7 +14,7 @@ def main():
     stdout, stderr = sphinx.communicate()
 
     if sphinx.returncode != 0:
-        raise subprocess.CalledProcessError(sphinx.returncode, cmd, output=stdout, stderr=stderr)
+        raise subprocess.CalledProcessError(sphinx.returncode, cmd, output=stdout + stderr)
```

Comment:
Good catch. I hadn't noticed that stderr were added in Python 3.5. In fact output was added in Python 2.7.

We shouldn't even be using an exception here, since it isn't going to be caught. Without a way to pass multi-line errors back to ansible-test this is about the best we can do (in case someone runs the smoke test directly):

```python
    if sphinx.returncode != 0:
        print('Command "%s" failed with status code: %d' % (' '.join(cmd), sphinx.returncode))
        print(stdout)
        print(stderr)
        return
```

Reply:
Well, that's what I have been doing locally for #45805 because it did not expose the information that I needed. So yes, I prefer this than local modifications for my own purpose only. Will do !

Reply:
https://github.com/ansible/ansible/pull/45958 updates ansible-test to show all the program output that did not match the regex instead of only the current line. I still need to improve the handling of unexpected failures, but at least this way you don't need to run the script directly to see what the output was.